### PR TITLE
make conda installs in CI stricter

### PR DESCRIPTION
**Pull request by jameslamb**
## Description

Contributes to https://github.com/rapidsai/build-planning/issues/106

Proposes specifying the package version in `conda install` calls that install CI artifacts, to reduce the risk of CI jobs picking up artifacts from other releases.

### CODE REVIEW

File: ci/test_conda.sh [hunk 1]
```diff
@@ -5,6 +5,8 @@ set -euo pipefail
 
 . /opt/conda/etc/profile.d/conda.sh
 
+VERSION="$(head -1 pynvjitlink/VERSION)"
```
Comment (jameslamb) [inline ci/test_conda.sh:8]: ref: https://github.com/rapidsai/pynvjitlink/blob/14639d8ffa6d67f6aa72a09991b9c125f277933b/conda/recipes/pynvjitlink/meta.yaml#L4-L7

**Review by bdice — APPROVED** [review]
Thanks @jameslamb. I think this is ready to merge.

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**